### PR TITLE
81 implement guild template api

### LINKED ISSUE #81: Implement Guild Template API
## As a user I want to...
be able to call the [guild template](https://discord.com/developers/docs/resources/guild-template) endpoints

## So that...
I can perform guild template actions using my discord bot

## Context
Discord provides a rich set of APIs for interacting with guild template in a discord server. We should implement all these endpoints. This ticket covers only the guild template endpoints. https://discord.com/developers/docs/resources/guild-template

## Acceptance Criteria
GIVEN a user wants to send a request to interact with a guild template on discord
THEN the specified action should be executed with sufficent error logging in place

### PR DESCRIPTION
**Pull request by notyourhabibti**
This pull requests introduces the guild template API.

closes #81

### CODE REVIEW

**Review by surajkumar — APPROVED** [review]

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/CreateGuildFromTemplateRequest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.javadiscord.jdi.internal.api.guild_template;
+
+import com.javadiscord.jdi.internal.api.DiscordRequest;
+import com.javadiscord.jdi.internal.api.DiscordRequestBuilder;
+
+import java.util.Map;
+
+public record CreateGuildFromTemplateRequest(String templateCode, String name, String icon)
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/CreateGuildFromTemplateRequest.java:8]: icon is optional

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplateRequest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.javadiscord.jdi.internal.api.guild_template;
+
+import com.javadiscord.jdi.internal.api.DiscordRequest;
+import com.javadiscord.jdi.internal.api.DiscordRequestBuilder;
+
+import java.util.Map;
+
+public record DeleteGuildTemplateRequest(
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplateRequest.java:8]: do u mean ModifyGuildTemplateRequest?

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplate.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.javadiscord.jdi.internal.api.guild_template;
+
+import com.javadiscord.jdi.internal.api.DiscordRequest;
+import com.javadiscord.jdi.internal.api.DiscordRequestBuilder;
+
+public record DeleteGuildTemplate(long guildId, String templateCode) implements DiscordRequest {
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplate.java:6]: postfix with request

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplateRequest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        long guildId, String templateCode, String name, String description)
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/DeleteGuildTemplateRequest.java:9]: name and description is optional

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/CreateGuildTemplateRequest.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.javadiscord.jdi.internal.api.guild_template;
+
+import com.javadiscord.jdi.internal.api.DiscordRequest;
+import com.javadiscord.jdi.internal.api.DiscordRequestBuilder;
+
+import java.util.Map;
+
+public record CreateGuildTemplateRequest(long guildId, String name, String description)
```
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild_template/CreateGuildTemplateRequest.java:8]: description is optional

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Review by surajkumar — APPROVED** [review]

File: api/src/main/java/com/javadiscord/jdi/internal/api/channel/StartThreadInForumOrMediaChannelRequest.java
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/channel/StartThreadInForumOrMediaChannelRequest.java:1]: revert

File: api/src/main/java/com/javadiscord/jdi/internal/api/guild/GetGuildWidgetImageRequest.java
Comment (Taz03) [inline api/src/main/java/com/javadiscord/jdi/internal/api/guild/GetGuildWidgetImageRequest.java:1]: revert

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/AllowedMentionType.java
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/AllowedMentionType.java:1]: revert

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/AllowedMentions.java
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/AllowedMentions.java:1]: revert

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/ForumMediaThreadMessageParam.java
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/ForumMediaThreadMessageParam.java:1]: revert

File: models/src/main/java/com/javadiscord/jdi/core/models/guild/WidgetStyle.java
Comment (Taz03) [inline models/src/main/java/com/javadiscord/jdi/core/models/guild/WidgetStyle.java:1]: revert

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Comment by surajkumar** [comment]
Closed in favour of https://github.com/javadiscord/java-discord-api/pull/122

**Pull request closed without merging**